service/raft/raft_rpc.cc

```diff
@@ -30,135 +32,99 @@ raft_rpc::raft_rpc(raft_state_machine& sm, netw::messaging_service& ms,
     , _address_map(address_map)
 {}
 
-future<raft::snapshot_reply> raft_rpc::send_snapshot(raft::server_id id, const raft::install_snapshot& snap, seastar::abort_source& as) {
-    return ser::raft_rpc_verbs::send_raft_send_snapshot(&_messaging,
-        netw::msg_addr(_address_map.get_inet_address(id)), db::no_timeout, _group_id, _server_id, id, snap);
-}
-
-future<> raft_rpc::send_append_entries(raft::server_id id, const raft::append_request& append_request) {
-    return ser::raft_rpc_verbs::send_raft_append_entries(&_messaging,
-        netw::msg_addr(_address_map.get_inet_address(id)), db::no_timeout, _group_id, _server_id, id, append_request);
-}
 
-void raft_rpc::send_append_entries_reply(raft::server_id id, const raft::append_reply& reply) {
-    (void)with_gate(_shutdown_gate, [this, id, &reply] {
-        return ser::raft_rpc_verbs::send_raft_append_entries_reply(&_messaging, netw::msg_addr(_address_map.get_inet_address(id)), timeout(), _group_id, _server_id, id, reply)
-            .handle_exception([id] (std::exception_ptr ex) {
+template <typename Verb, typename Msg> void
+raft_rpc::one_way_rpc(std::experimental::source_location loc, raft::server_id id,
+        Verb&& verb, Msg&& msg) {
+    (void)with_gate(_shutdown_gate, [this, loc = std::move(loc), id, &verb, &msg] {
+        auto ip_addr = _address_map.find(id);
+        if (!ip_addr) {
+            rlogger.debug("{}:{}: {} dropping outgoing message to {} - IP address not found",
+                loc.file_name(), loc.line(), loc.function_name(), id);
+            return make_ready_future<>();
+        }
+        return verb(&_messaging, netw::msg_addr(*ip_addr), timeout(), _group_id, _server_id, id, std::forward<Msg>(msg))
+            .handle_exception([loc = std::move(loc), id] (std::exception_ptr ex) {
```

Comment:
It's a bunch of sstrings.

Reply:
I think it's not specified what exactly they are. A string could be `std::string` (expensive copies) or it could be `std::string_view` (cheap copies).

(Although I doubt `source_location` performs an allocation, it probably is just a pointer to a string living somewhere in .rodata or similar)

But let's stay on the safe side and ensure they are always moved.